### PR TITLE
Issue 4 - Tests for within strata assignments

### PR DESCRIPTION
**Pull request by redouane-dziri**
More tests to check the proportions of assignments within strata

### CODE REVIEW

File: tests/test_stochatreat.py [hunk 1]
```diff
@@ -277,6 +277,135 @@ def test_stochatreat_only_misfits(probs):
...
+        data=df,
+        block_cols=block_cols,
+        treats=len(probs),
+        idx_col="id",
+        probs=probs,
+        random_state=42,
+    )
+    counts_diff = compute_counts_diff(treats, probs)
+
+    np.testing.assert_equal(
```
Comment (RoyalTS) [inline tests/test_stochatreat.py]: I think `not any()` is the more idiomatic way of doing that.

File: tests/test_stochatreat.py [hunk 2]
```diff
@@ -277,6 +277,135 @@ def test_stochatreat_only_misfits(probs):
...
+        (np.absolute(counts_diff) > 1).sum(),
+        0
+    )
+
+
+@pytest.mark.parametrize("probs", standard_probs)
+def test_stochatreat_within_strata_only_misfits(probs):
```
Comment (RoyalTS) [inline tests/test_stochatreat.py]: depending on how you assign the misfits this need not or maybe even should not be true, right?

**Review by RoyalTS — CHANGES_REQUESTED** [review]

File: tests/test_stochatreat.py (hunk 2 above)
Comment (redouane-dziri) [reply tests/test_stochatreat.py]: Yes you're right should rather test if the sum of counts diff within each block is always less than the number of treatments (i.e len(probs) for example), right?

**Comment by RoyalTS** [comment]
Did these ever all pass for you?

**Pull request merged**